agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java

```diff
@@ -743,7 +743,8 @@ static void overlayFromEnv(
     String replacedConnectionString = stringSubstitutor.replace(config.connectionString);
     if (replacedConnectionString != null
         && !replacedConnectionString.startsWith("InstrumentationKey=")
-        && config.connectionString.equals(replacedConnectionString)) {
+        && config.connectionString.equals(replacedConnectionString)
+        && System.getenv(APPLICATIONINSIGHTS_CONNECTION_STRING_ENV) == null) {
```

Comment:
maybe it's better to check env var before loading from the file? 

It's not needed to load from the file when env var is present.

Reply:
To do this with a code that is easy to read,  https://github.com/microsoft/ApplicationInsights-Java/blob/5481277381728dcbc985117da619e61225be2237/agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java#L757C9-L757C39 should be called before loading from a file. It seems that it would require to change several lines of code and check carefully that it does not break anything. In this PR, I'd rather make the minimal change and be confident to not break something. An unecessary read from a file may not be a user issue.